include/rdma/fi_domain.h

```diff
@@ -128,6 +128,7 @@ enum fi_hmem_iface {
 	FI_HMEM_ZE,
 	FI_HMEM_NEURON,
 	FI_HMEM_SYNAPSEAI,
+	FI_NUM_HMEM_TYPES, /* Keep this flag last to calculate the length of hmem_ops*/
```

Comment:
@darrylabbate uses `(sizeof efa_hmem_ifaces)` which I think is a little cleaner

Reply:
That only works since `efa_hmem_ifaces` is an array of `fi_hmem_iface` (which I created for my PR). C doesn't offer any way to easily retrieve the number of values defined in some enum.

Reply:
what you would want is this:

```c
#define FI_HMEM_IFACE_FOREACH(var) \
	for ((var) = 0; (var) < ((sizeof hmem_ops) / (sizeof (struct ofi_hmem_ops))); ++(var))

#define FF_HMEM_IFACE_FOREACH_NON_SYSTEM(var) \
	for ((var) = 1; (var) < ((sizeof hmem_ops) / (sizeof (struct ofi_hmem_ops))); ++(var))
```

Reply:
The macro would need to be defined in the .h, and it wouldn't work because `hmem_ops` is defined as `struct ofi_hmem_ops hmemops [] = {x, y, z}` in the .c, which leads to a compile issue of not being able to known the size of the array at compile time.